packages/react/src/components/Table/TableToolbar/TableToolbar.jsx

```diff
@@ -305,9 +311,21 @@ const TableToolbar = ({
               translateWithId={(...args) => tableTranslateWithId(i18n, ...args)}
               id={`${tableId}-toolbar-search`}
               onChange={(event, defaultValue) => {
-                // https://github.com/carbon-design-system/carbon/issues/6157
-                onApplySearch(event?.target?.value || defaultValue);
+                const value = event?.target?.value || (defaultValue ?? '');
+                if (hasFastSearch) {
+                  onApplySearch(value);
+                }
               }}
+              onKeyDown={
+                hasFastSearch
+                  ? undefined
+                  : handleSpecificKeyDown(['Enter'], (e) => onApplySearch(e.target.value))
+              }
+              onClear={() => onApplySearch('')}
+              // This can't be used yet b/c it prevents the search box from automatically
+              // closing on blur.
+              // https://github.com/carbon-design-system/carbon/issues/10077
```

Comment:
I think it is better to create a PAL issue linking to the carbon issue instead of documenting it in the code. Same goes for the comments in the Table.test.jsx

Reply:
I opened a PAL issue, too, but I don't see the reason to remove these comments and make someone re-create the tests later once the other issue is resolved. I've already checked them and they work, but it just breaks the auto-expand/contract of the search box.